From 06fe7ee6d2297e9b63daae1b73e36816c6736c37 Mon Sep 17 00:00:00 2001
From: Xiaokang Qin <xiaokang.qin@intel.com>
Date: Wed, 14 Nov 2012 11:21:33 +0800
Subject: allow for custom Recovery, TFP, and otapackage rules

For recovery and ota stuff, introduce conditional compilation
so that it is possible to override them in a product-specific
makefile (such as under vendor/*/build/tasks)

This allows for customized rules to create boot images,
Target Files Packages, and OTA updates when using non-AOSP
boot images and/or custom OTA tools.

Bug: ADEV-129
Category: Device Enablement
Domain: AOSP-build, AOSP-recovery
Severity: High
Origin: Internal
Upstream-Candidate: Yes

Change-Id: I81a8e1087ca2f6f9d831ebff33e32660b13c8caa
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
Signed-off-by: Xiaokang Qin <xiaokang.qin@intel.com>
---
 core/Makefile |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 74d52ff..3c2fc20 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -679,6 +679,8 @@ endef
 # -----------------------------------------------------------------
 # Recovery image
 
+ifneq ($(TARGET_MAKE_NO_DEFAULT_RECOVERY),true)
+
 # If neither TARGET_NO_KERNEL nor TARGET_NO_RECOVERY are true
 ifeq (,$(filter true, $(TARGET_NO_KERNEL) $(TARGET_NO_RECOVERY)))
 
@@ -823,6 +825,7 @@ endif
 
 .PHONY: recoveryimage
 recoveryimage: $(INSTALLED_RECOVERYIMAGE_TARGET) $(RECOVERY_RESOURCE_ZIP)
+endif # TARGET_MAKE_NO_DEFAULT_RECOVERY
 
 ifneq ($(BOARD_NAND_PAGE_SIZE),)
 mkyaffs2_extra_flags := -c $(BOARD_NAND_PAGE_SIZE)
@@ -1188,6 +1191,8 @@ otatools: $(OTATOOLS)
 # This zip can be used to create an OTA package or filesystem image
 # as a post-build step.
 #
+ifneq ($(TARGET_MAKE_NO_DEFAULT_OTA_PACKAGE),true)
+
 name := $(TARGET_PRODUCT)
 ifeq ($(TARGET_BUILD_TYPE),debug)
   name := $(name)_debug
@@ -1408,6 +1413,7 @@ endif    # recovery_fstab is defined
 endif    # TARGET_NO_KERNEL != true
 endif    # TARGET_DEVICE != generic*
 endif    # TARGET_PRODUCT != sdk
+endif # TARGET_MAKE_NO_DEFAULT_OTA_PACKAGE
 
 # -----------------------------------------------------------------
 # A zip of the symbols directory.  Keep the full paths to make it
-- 
1.7.4.1

